src/parsec_client/api_overview.md

```diff
@@ -177,6 +187,10 @@ permitted numerical values for this field are given as follows:-
 - A value of 2 (`0x02`) indicates authentication tokens. The service will expect the
    **authentication** field to contain a JWT token. Tokens must be signed with the private key of
    the identity provider and their validity period must cover the moment when the check is done.
+- A value of 3 (`0x03`) indicates peer credentials authentication. The service will expect the
+   **authentication** field to contain the stringified UID of the connecting process. The Parsec
```

Comment:
"stringified uid" feels a bit vague here considering this section is supposed to be the specification of how a client process should populate the auth field. We don't mention that it has to be UTF8 (although that is mentioned somewhere further down, but I think it needs to be here). We also probably want to be very explicit that we expect a UID and **not** a user name. Also, since UIDs are numerical, we probably want to specify what number format we expect. Decimal? Hex? Will the service parse conventional notations such as the` 0x` prefix for hex numbers? This is the level of precision that I think we need here.

Reply:
Good point -- I've changed this to:

```
- A value of 3 (`0x03`) indicates peer credentials authentication. The service will expect the                                                                                                                                                                                                                                                                                                                                           
  **authentication** field to contain the non-padded decimal UID (**not** username) of the
  connecting process as a UTF-8 string (e.g. `123`). The Parsec service will verify that this
  self-declared UID is consistent with the UID from the peer credentials.
```

Please let me know if there's any further suggestions!

Reply:
I think to answer to your original question about the format, we need to be clear about the scope of the authentictor. Is it a generic peer credentials authenticator which could be used with other operating systems/listener types or a **Unix** peer credential authenticator?
If the authenticator that we are describing here is only the later then I think we should:
* be more precise where we mention "peer credentials" and always say "Unix peer credentials" and make sure that when we say "UID" we mean the specific Unix User identifier
* the Unix User Identifier is represented with [an integer type](https://en.wikipedia.org/wiki/User_identifier#Type). Most Unix OS use 32 bits for this, and Linux uses the `uid_t` type which is an unsigned 32-bits number. For that reason, I think it would be better to expect the authentication payload to be the representation of a `u32` value, 4-bytes padded with zeroes. For example the UID `4` would be `[0x00, 0x00, 0x00, 0x04]` (or reverse depending on our endianess). I think this is more efficient and less error-prone than having to parse/format a UTF-8 string.

Reply:
> I think to answer to your original question about the format, we need to be clear about the scope of the authentictor. Is it a generic peer credentials authenticator which could be used with other operating systems/listener types or a Unix peer credential authenticator?

AFAIK 'peer credentials' is a Unix-exclusive thing, but I'm all for being more explicit. 😄 

> be more precise where we mention "peer credentials" and always say "Unix peer credentials" and make sure that when we say "UID" we mean the specific Unix User identifier

ACK, will do.

> the Unix User Identifier is represented with an integer type. Most Unix OS use 32 bits for this, and Linux uses the uid_t type which is an unsigned 32-bits number. For that reason, I think it would be better to expect the authentication payload to be the representation of a u32 value, 4-bytes padded with zeroes. For example the UID 4 would be [0x00, 0x00, 0x00, 0x04] (or reverse depending on our endianess). I think this is more efficient and less error-prone than having to parse/format a UTF-8 string.

Agreed, that's a lot nicer. Will do. 🙂